packages/ibm-products/src/components/Datagrid/Datagrid.test.js

```diff
@@ -1379,26 +1411,21 @@ describe(componentName, () => {
     const rowExpander = row.querySelector(`button[aria-label="Expand row"]`);
     fireEvent.click(rowExpander);
 
-    expect(
-      screen
-        .getByRole('table')
-        .getElementsByTagName('tbody')[0]
-        .getElementsByClassName('c4p--datagrid__expanded-row')
-    ).toBeDefined();
-    expect(
-      screen
-        .getByRole('table')
-        .getElementsByTagName('tbody')[0]
-        .getElementsByClassName('c4p--datagrid__expanded-row')[0].textContent
-    ).toEqual(`Content for ${rowNumber}`);
+    expect(row.nextElementSibling).toHaveClass(`${blockClass}__expanded-row`);
+    expect(row.nextElementSibling.textContent).toEqual(
+      `Content for ${rowNumber}`
+    );
+
+    fireEvent.mouseOver(row.nextElementSibling);
+    fireEvent.mouseLeave(row.nextElementSibling);
 
     const rowExpanderCollapse = row.querySelector(
       `button[aria-label="Collapse row"]`
     );
     fireEvent.click(rowExpanderCollapse);
   }
 
-  it('Expanded Row', async () => {
+  it('Expanded Row', () => {
```

Comment:
Maybe a future change that we can open in a new issue — but it would probably be nice if we were clearer about what’s being tested in each test rather than just naming the extensions.

Reply:
Included as part of https://github.com/carbon-design-system/ibm-products/pull/3813